src/tools/gophish_export.py

```diff
@@ -253,6 +253,44 @@ def get_application(rawEvent):
     return application
 
 
+def export_user_reports(api, assessment_id):
+    """Build and export a user_report JSON object for each campaign in an assessment."""
+    campaign_ids = get_campaign_ids(api, assessment_id)
+
+    for campaign_id in campaign_ids:
+        first_report = None
+        user_report_doc = dict()
+        campaign = get_campaign_data(api, campaign_id)
+
+        # iterate over clicks and find the earliest click
+        for click in campaign["clicks"]:
+            if (
+                first_report is None
+                or datetime.strptime(click["time"].split(".")[0], "%Y-%m-%dT%H:%M:%S")
+                < first_report
+            ):
+                first_report = datetime.strptime(
+                    click["time"].split(".")[0], "%Y-%m-%dT%H:%M:%S"
+                )
+
+        user_report_doc["customer"] = None
+        user_report_doc["assessment"] = assessment_id
+        user_report_doc["campaign"] = campaign["id"]
+        user_report_doc["first_report"] = datetime.strftime(
+            first_report, "%Y-%m-%dT%H:%M:%S"
+        )
+        user_report_doc["total_num_reports"] = api.campaigns.summary(
+            campaign_id=campaign_id
+        ).stats.clicked
+
+        logging.info(
+            f"Writing out user reports for all campaigns in assessment: {assessment_id}"
```

Comment:
Since this logging message is inside the `for campaign_id in campaign_ids` loop, it should indicate which campaign is being output:
```suggestion
            f"Writing out user report for campaign {campaign_id} in assessment {assessment_id}"
```